// vocs/_lib/vite/.vocs/dist/assets/actions-v5NXZGFv.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Actions",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    code: "code",
    div: "div",
    h1: "h1",
    h2: "h2",
    h3: "h3",
    header: "header",
    li: "li",
    ol: "ol",
    p: "p",
    pre: "pre",
    span: "span",
    strong: "strong",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "actions",
        children: ["Actions", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#actions",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.p, {
      children: "Actions are strings returned by a node's Post phase that determine the next step in a workflow. They provide dynamic, runtime control over flow execution."
    }), "\n", jsxs(_components.h2, {
      id: "how-actions-work",
      children: ["How Actions Work", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#how-actions-work",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "When a node completes, its Post phase returns an action:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "func"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " ("
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "n "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "*"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "MyNode"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ") "
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "Post"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "ctx"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: " context"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "Context"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "shared"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " *"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "flyt"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "SharedStore"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "prepResult"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "execResult"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: " any"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ") ("
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "flyt"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "Action"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "error"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ") {"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "    if"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " success {"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "        return"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: ' "continue"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "nil"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    }"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "    return"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: ' "retry"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "nil"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "}"
            })
          })]
        })
      })
    }), "\n", jsx(_components.p, {
      children: "The flow then routes to the node connected with that action:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "flow "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: ":="
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " flyt."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "NewFlow"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "(startNode)"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "flow."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "Connect"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "(startNode, "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"continue"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", processNode)"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "flow."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "Connect"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "(startNode, "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"retry"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", retryNode)"
            })]
          })]
        })
      })
    }), "\n", jsxs(_components.h2, {
      id: "default-action",
      children: ["Default Action", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#default-action",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["The most common action is ", jsx(_components.code, {
        children: "flyt.DefaultAction"
      }), ":"]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "// Using the constant"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "return"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " flyt.DefaultAction, "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "nil"
            }), jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: '  // Returns "default"'
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "// In flow connections"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "flow."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "Connect"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "(node1, flyt.DefaultAction, node2)"
            })]
          })]
        })
      })
    }), "\n", jsxs(_components.h2, {
      id: "action-based-routing",
      children: ["Action-Based Routing", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#action-based-routing",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "simple-branching",
      children: ["Simple Branching", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#simple-branching",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "validationNode "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: ":="
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " flyt."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "NewNode"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    flyt."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "WithExecFunc"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "func"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "ctx"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: " context"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "Context"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "prepResult"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: " any"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ") ("
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "any"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "error"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ") {"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "        data "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: ":="
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " prepResult.("
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "string"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ")"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "        return"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: " len"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "(data) "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: ">"
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: " 0"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "nil"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    }),"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    flyt."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "WithPostFunc"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "func"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "ctx"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: " context"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "Context"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "shared"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " *"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "flyt"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "SharedStore"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "prepResult"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "execResult"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: " any"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ") ("
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "flyt"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "Action"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "error"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ") {"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "        if"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " execResult.("
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "bool"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ") {"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "            return"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: ' "valid"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "nil"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "        }"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "        return"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: ' "invalid"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "nil"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    }),"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ")"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "flow "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: ":="
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " flyt."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "NewFlow"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "(validationNode)"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "flow."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "Connect"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "(validationNode, "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"valid"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", processNode)"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "flow."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "Connect"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "(validationNode, "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"invalid"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", errorNode)"
            })]
          })]
        })
      })
    }), "\n", jsxs(_components.h3, {
      id: "multi-way-branching",
      children: ["Multi-Way Branching", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#multi-way-branching",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "categoryNode "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: ":="
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " flyt."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "NewNode"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    flyt."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "WithExecFunc"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "func"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "ctx"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: " context"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "Context"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "prepResult"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: " any"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ") ("
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "any"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "error"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ") {"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "        value "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: ":="
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " prepResult.("
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "int"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ")"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "        if"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " value "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "<"
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: " 10"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " {"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "            return"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: ' "small"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "nil"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "        } "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "else"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " if"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " value "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "<"
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: " 100"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " {"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "            return"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: ' "medium"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "nil"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "        }"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "        return"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: ' "large"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "nil"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    }),"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    flyt."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "WithPostFunc"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "func"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "ctx"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: " context"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "Context"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "shared"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " *"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "flyt"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "SharedStore"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "prepResult"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "execResult"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: " any"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ") ("
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "flyt"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "Action"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "error"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ") {"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "        return"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " flyt."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "Action"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "(execResult.("
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "string"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ")), "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "nil"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    }),"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ")"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "flow "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: ":="
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " flyt."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "NewFlow"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "(categoryNode)"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "flow."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "Connect"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "(categoryNode, "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"small"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", smallHandler)"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "flow."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "Connect"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "(categoryNode, "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"medium"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", mediumHandler)"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "flow."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "Connect"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "(categoryNode, "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"large"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", largeHandler)"
            })]
          })]
        })
      })
    }), "\n", jsxs(_components.h2, {
      id: "flow-termination",
      children: ["Flow Termination", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#flow-termination",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "If no connection exists for an action, the flow terminates:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "flow "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: ":="
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " flyt."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "NewFlow"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "(node1)"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "flow."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "Connect"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "(node1, "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"continue"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", node2)"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: '// If node1 returns "stop", flow ends (no connection for "stop")'
            })
          })]
        })
      })
    }), "\n", jsx(_components.p, {
      children: "Explicitly terminate by connecting to nil:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsx(_components.code, {
          children: jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "flow."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "Connect"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "(finalNode, "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"done"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "nil"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ")"
            })]
          })
        })
      })
    }), "\n", jsxs(_components.h2, {
      id: "common-patterns",
      children: ["Common Patterns", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#common-patterns",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "successfailure-pattern",
      children: ["Success/Failure Pattern", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#successfailure-pattern",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "func"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: " makeDecisionPost"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "ctx"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: " context"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "Context"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "shared"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " *"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "flyt"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "SharedStore"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "prepResult"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "execResult"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: " any"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ") ("
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "flyt"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "Action"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "error"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ") {"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "    if"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " err "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: ":="
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " execResult.("
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "error"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "); err "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "!="
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: " nil"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " {"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "        return"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: ' "failure"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "nil"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    }"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "    return"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: ' "success"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "nil"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "}"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.h3, {
      id: "retry-pattern",
      children: ["Retry Pattern", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#retry-pattern",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "func"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: " retryablePost"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "ctx"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: " context"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "Context"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "shared"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " *"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "flyt"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "SharedStore"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "prepResult"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "execResult"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: " any"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ") ("
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "flyt"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "Action"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "error"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ") {"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    attempts, _ "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: ":="
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " shared."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "Get"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"attempts"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ")"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    count "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: ":="
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " attempts.("
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "int"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ")"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    "
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "    if"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " execResult "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "=="
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: " nil"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " &&"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " count "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "<"
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: " 3"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " {"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "        shared."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "Set"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"attempts"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", count "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "+"
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: " 1"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ")"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "        return"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: ' "retry"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "nil"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    }"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "    return"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: ' "continue"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "nil"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "}"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.h3, {
      id: "state-machine-pattern",
      children: ["State Machine Pattern", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#state-machine-pattern",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "func"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: " statePost"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "ctx"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: " context"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "Context"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "shared"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " *"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "flyt"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "SharedStore"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "prepResult"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "execResult"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: " any"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ") ("
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "flyt"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "Action"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "error"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ") {"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    currentState, _ "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: ":="
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " shared."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "Get"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"state"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ")"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    "
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "    switch"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " currentState.("
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "string"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ") {"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "    case"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: ' "init"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ":"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "        shared."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "Set"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"state"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"processing"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ")"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "        return"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: ' "process"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "nil"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "    case"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: ' "processing"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ":"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "        shared."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "Set"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"state"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"complete"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ")"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "        return"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: ' "finalize"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "nil"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "    default"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ":"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "        return"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: ' "done"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "nil"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    }"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "}"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.h2, {
      id: "dynamic-actions",
      children: ["Dynamic Actions", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#dynamic-actions",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Actions can be computed at runtime:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "func"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: " dynamicPost"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "ctx"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: " context"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "Context"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "shared"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " *"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "flyt"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "SharedStore"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "prepResult"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "execResult"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: " any"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ") ("
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "flyt"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "Action"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "error"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ") {"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    result "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: ":="
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " execResult.("
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "map"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "["
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "string"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "]"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "any"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ")"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    nextAction "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: ":="
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " result["
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"next_action"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "].("
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "string"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ")"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "    return"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " flyt."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "Action"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "(nextAction), "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "nil"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "}"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.h2, {
      id: "best-practices",
      children: ["Best Practices", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#best-practices",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Use Descriptive Names"
        }), ": Actions should clearly indicate their purpose"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Document Actions"
        }), ": List all possible actions a node can return"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Handle All Cases"
        }), ": Ensure all possible actions have connections or intentionally terminate"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Consistent Naming"
        }), ": Use a consistent naming scheme across your application"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Avoid Magic Strings"
        }), ": Define action constants for reusability"]
      }), "\n"]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "const"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " ("
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "    ActionValidate"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " ="
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " flyt."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "Action"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"validate"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ")"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "    ActionProcess"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "  ="
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " flyt."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "Action"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"process"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ")"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "    ActionComplete"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " ="
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " flyt."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "Action"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"complete"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ")"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "    ActionError"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "    ="
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " flyt."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "Action"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"error"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ")"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ")"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.h2, {
      id: "next-steps",
      children: ["Next Steps", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#next-steps",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.a, {
          href: "/concepts/flows",
          children: "Flows"
        }), " - Connect nodes with actions"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.a, {
          href: "/patterns/branching",
          children: "Conditional Branching"
        }), " - Advanced routing patterns"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.a, {
          href: "/advanced/nested-flows",
          children: "Nested Flows"
        }), " - Compose complex workflows"]
      }), "\n"]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
