// vocs/_lib/vite/.vocs/dist/assets/index-DXdzbOqL.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
import "react";
import { L as Link, b as button$1, a as button_accent, u as useConfig, c as Logo$1, R as Root$1, d as List, T as Trigger, C as Content } from "../index.server.js";
import clsx from "clsx";
import "react-router";
import "node:fs";
import "node:path";
import "toml";
import "vite";
import "@vanilla-extract/dynamic";
import "react-intersection-observer";
import "@mdx-js/mdx";
import "@radix-ui/react-icons";
import "@radix-ui/react-dialog";
import "minisearch";
import "@radix-ui/react-label";
import "mark.js";
import "@radix-ui/react-navigation-menu";
import "@radix-ui/react-accordion";
import "radix-ui";
import "@radix-ui/react-popover";
import "@mdx-js/react";
import "@radix-ui/react-tabs";
import "@floating-ui/react";
function Button$1({ children, className, href, variant }) {
  return jsx(Link, { className: clsx(className, button$1, variant === "accent" && button_accent), href, variant: "styleless", children });
}
var root = "vocs_HomePage";
var logo = "vocs_HomePage_logo";
var title = "vocs_HomePage_title";
var tagline = "vocs_HomePage_tagline";
var description = "vocs_HomePage_description";
var buttons = "vocs_HomePage_buttons";
var button = "vocs_HomePage_button";
var tabs = "vocs_HomePage_tabs";
var tabsList = "vocs_HomePage_tabsList";
var tabsContent = "vocs_HomePage_tabsContent";
var packageManager = "vocs_HomePage_packageManager";
function Root({ children, className }) {
  return jsx("div", { className: clsx(className, root), children });
}
function Logo({ className }) {
  const { logoUrl, title: title$1 } = useConfig();
  return logoUrl ? jsx("div", { className: clsx(className, logo), children: jsx(Logo$1, {}) }) : jsx("h1", { className: clsx(className, title), children: title$1 });
}
function Tagline({ children, className }) {
  return jsx("div", { className: clsx(className, tagline), children });
}
function Description({ children, className }) {
  return jsx("div", { className: clsx(className, description), children });
}
function Buttons({ children, className }) {
  return jsx("div", { className: clsx(className, buttons), children });
}
function Button(props) {
  return jsx(Button$1, { ...props, className: clsx(button, props.className) });
}
function InstallPackage({ name, type = "install" }) {
  return jsxs(Root$1, { className: tabs, defaultValue: "npm", children: [jsxs(List, { className: tabsList, children: [jsx(Trigger, { value: "npm", children: "npm" }), jsx(Trigger, { value: "pnpm", children: "pnpm" }), jsx(Trigger, { value: "yarn", children: "yarn" })] }), jsxs(Content, { className: tabsContent, value: "npm", children: [jsx("span", { className: packageManager, children: "npm" }), " ", type === "init" ? "init" : "install", " ", name] }), jsxs(Content, { className: tabsContent, value: "pnpm", children: [jsx("span", { className: packageManager, children: "pnpm" }), " ", type === "init" ? "create" : "add", " ", name] }), jsxs(Content, { className: tabsContent, value: "yarn", children: [jsx("span", { className: packageManager, children: "yarn" }), " ", type === "init" ? "create" : "add", " ", name] })] });
}
const HomePage = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  Button,
  Buttons,
  Description,
  InstallPackage,
  Logo,
  Root,
  Tagline
}, Symbol.toStringTag, { value: "Module" }));
const frontmatter = {
  "layout": "landing"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    code: "code",
    div: "div",
    h2: "h2",
    p: "p",
    pre: "pre",
    span: "span",
    ...useMDXComponents(),
    ...props.components
  };
  if (!HomePage) _missingMdxReference("HomePage", false);
  if (!Button) _missingMdxReference("HomePage.Button", true);
  if (!Buttons) _missingMdxReference("HomePage.Buttons", true);
  if (!Description) _missingMdxReference("HomePage.Description", true);
  if (!Root) _missingMdxReference("HomePage.Root", true);
  if (!Tagline) _missingMdxReference("HomePage.Tagline", true);
  return jsxs(Fragment, {
    children: [jsxs(Root, {
      children: [jsxs("div", {
        style: {
          display: "flex",
          flexDirection: "column",
          alignItems: "center",
          marginBottom: "1rem"
        },
        children: [jsx("img", {
          src: "/flyt-logo.png",
          alt: "Flyt Logo",
          style: {
            width: "280px",
            height: "auto"
          }
        }), jsx("p", {
          style: {
            fontStyle: "italic",
            color: "var(--vocs-color_text2)",
            marginTop: "0.5rem",
            fontSize: "0.95rem"
          },
          children: jsx(_components.p, {
            children: 'Norwegian for "flow" • Pronounced "fleet"'
          })
        })]
      }), jsx(Tagline, {
        children: "A minimalist workflow framework for Go"
      }), jsx(Fragment, {
        children: jsx(_components.pre, {
          className: "shiki shiki-themes github-light github-dark-dimmed",
          style: {
            backgroundColor: "#fff",
            "--shiki-dark-bg": "#22272e",
            color: "#24292e",
            "--shiki-dark": "#adbac7"
          },
          tabIndex: "0",
          children: jsx(_components.code, {
            children: jsxs(_components.span, {
              className: "line",
              children: [jsx(_components.span, {
                style: {
                  color: "#6F42C1",
                  "--shiki-dark": "#F69D50"
                },
                children: "go"
              }), jsx(_components.span, {
                style: {
                  color: "#032F62",
                  "--shiki-dark": "#96D0FF"
                },
                children: " get"
              }), jsx(_components.span, {
                style: {
                  color: "#032F62",
                  "--shiki-dark": "#96D0FF"
                },
                children: " github.com/mark3labs/flyt"
              })]
            })
          })
        })
      }), jsx(Description, {
        children: jsx(_components.p, {
          children: "Build powerful, composable workflows with zero dependencies. Flyt provides a simple yet flexible\ngraph-based abstraction for orchestrating tasks in Go."
        })
      }), jsxs(Buttons, {
        children: [jsx(Button, {
          href: "/getting-started/quick-start",
          variant: "accent",
          children: "Get Started"
        }), jsx(Button, {
          href: "https://github.com/mark3labs/flyt",
          children: "GitHub"
        })]
      })]
    }), "\n", jsxs("div", {
      style: {
        marginTop: "4rem"
      },
      children: [jsxs(_components.h2, {
        id: "features",
        children: ["Features", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#features",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      }), jsxs("div", {
        style: {
          display: "grid",
          gridTemplateColumns: "repeat(auto-fit, minmax(300px, 1fr))",
          gap: "1.5rem",
          marginTop: "2rem"
        },
        children: [jsxs("div", {
          children: [jsx("h3", {
            style: {
              fontSize: "1.1rem",
              marginBottom: "0.5rem"
            },
            children: "🚀 Zero Dependencies"
          }), jsx("p", {
            style: {
              color: "var(--vocs-color_text2)",
              fontSize: "0.95rem"
            },
            children: "Pure Go standard library - no external dependencies required"
          })]
        }), jsxs("div", {
          children: [jsx("h3", {
            style: {
              fontSize: "1.1rem",
              marginBottom: "0.5rem"
            },
            children: "🔄 Composable Workflows"
          }), jsx("p", {
            style: {
              color: "var(--vocs-color_text2)",
              fontSize: "0.95rem"
            },
            children: "Build complex flows from simple, reusable nodes"
          })]
        }), jsxs("div", {
          children: [jsx("h3", {
            style: {
              fontSize: "1.1rem",
              marginBottom: "0.5rem"
            },
            children: "🎯 Action-Based Routing"
          }), jsx("p", {
            style: {
              color: "var(--vocs-color_text2)",
              fontSize: "0.95rem"
            },
            children: "Dynamic flow control based on runtime results"
          })]
        }), jsxs("div", {
          children: [jsx("h3", {
            style: {
              fontSize: "1.1rem",
              marginBottom: "0.5rem"
            },
            children: "🔁 Built-in Retry Logic"
          }), jsx("p", {
            style: {
              color: "var(--vocs-color_text2)",
              fontSize: "0.95rem"
            },
            children: "Handle transient failures with configurable retries"
          })]
        }), jsxs("div", {
          children: [jsx("h3", {
            style: {
              fontSize: "1.1rem",
              marginBottom: "0.5rem"
            },
            children: "🔒 Thread-Safe"
          }), jsx("p", {
            style: {
              color: "var(--vocs-color_text2)",
              fontSize: "0.95rem"
            },
            children: "Concurrent batch processing with safe shared state"
          })]
        }), jsxs("div", {
          children: [jsx("h3", {
            style: {
              fontSize: "1.1rem",
              marginBottom: "0.5rem"
            },
            children: "📦 Batch Processing"
          }), jsx("p", {
            style: {
              color: "var(--vocs-color_text2)",
              fontSize: "0.95rem"
            },
            children: "Process collections efficiently with built-in patterns"
          })]
        })]
      })]
    }), "\n", jsxs("div", {
      style: {
        marginTop: "3rem"
      },
      children: [jsxs(_components.h2, {
        id: "quick-example",
        children: ["Quick Example", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#quick-example",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      }), jsx("div", {
        style: {
          marginTop: "1.5rem"
        },
        children: jsx(Fragment, {
          children: jsx(_components.pre, {
            className: "shiki shiki-themes github-light github-dark-dimmed",
            style: {
              backgroundColor: "#fff",
              "--shiki-dark-bg": "#22272e",
              color: "#24292e",
              "--shiki-dark": "#adbac7"
            },
            tabIndex: "0",
            children: jsxs(_components.code, {
              children: [jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "package"
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: " main"
                })]
              }), "\n", jsx(_components.span, {
                className: "line",
                "data-empty-line": true,
                children: " "
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "import"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " ("
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '    "'
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "context"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"'
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '    "'
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "fmt"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"'
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '    "'
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "github.com/mark3labs/flyt"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"'
                })]
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ")"
                })
              }), "\n", jsx(_components.span, {
                className: "line",
                "data-empty-line": true,
                children: " "
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "func"
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: " main"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "() {"
                })]
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#6A737D",
                    "--shiki-dark": "#768390"
                  },
                  children: "    // Create nodes that share data via SharedStore"
                })
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "    fetchNode "
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: ":="
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " flyt."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "NewNode"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "("
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "        flyt."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "WithExecFunc"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "("
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "func"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "("
                }), jsx(_components.span, {
                  style: {
                    color: "#E36209",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "ctx"
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: " context"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "Context"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#E36209",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "_"
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: " any"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ") ("
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "any"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "error"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ") {"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "            fmt."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "Println"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "("
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"Fetching data..."'
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ")"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "            return"
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: " map"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "["
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "string"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "]"
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "int"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "{"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"value"'
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ": "
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: "42"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "}, "
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: "nil"
                })]
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "        }),"
                })
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "        flyt."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "WithPostFunc"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "("
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "func"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "("
                }), jsx(_components.span, {
                  style: {
                    color: "#E36209",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "ctx"
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: " context"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "Context"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#E36209",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "shared"
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: " *"
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "flyt"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "SharedStore"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#E36209",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "_"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#E36209",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "execResult"
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: " any"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ") ("
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "flyt"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "Action"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "error"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ") {"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "            shared."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "Set"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "("
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"data"'
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", execResult)"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "            return"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " flyt.DefaultAction, "
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: "nil"
                })]
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "        }),"
                })
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "    )"
                })
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "    "
                })
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "    processNode "
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: ":="
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " flyt."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "NewNode"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "("
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "        flyt."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "WithPrepFunc"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "("
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "func"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "("
                }), jsx(_components.span, {
                  style: {
                    color: "#E36209",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "ctx"
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: " context"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "Context"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#E36209",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "shared"
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: " *"
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "flyt"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "SharedStore"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ") ("
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "any"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "error"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ") {"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "            data, _ "
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: ":="
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " shared."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "Get"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "("
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"data"'
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ")"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "            return"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " data, "
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: "nil"
                })]
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "        }),"
                })
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "        flyt."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "WithExecFunc"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "("
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "func"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "("
                }), jsx(_components.span, {
                  style: {
                    color: "#E36209",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "ctx"
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: " context"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "Context"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#E36209",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "data"
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: " any"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ") ("
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "any"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "error"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ") {"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "            val "
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: ":="
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " data.("
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "map"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "["
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "string"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "]"
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "int"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ")["
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"value"'
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "]"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "            fmt."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "Printf"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "("
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"Processing: '
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#F47067"
                  },
                  children: "%d\\n"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"'
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", val)"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "            return"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " val "
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "*"
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: " 2"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: "nil"
                })]
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "        }),"
                })
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "        flyt."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "WithPostFunc"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "("
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "func"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "("
                }), jsx(_components.span, {
                  style: {
                    color: "#E36209",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "ctx"
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: " context"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "Context"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#E36209",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "shared"
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: " *"
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "flyt"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "SharedStore"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#E36209",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "_"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#E36209",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "execResult"
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: " any"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ") ("
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "flyt"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "Action"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "error"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ") {"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "            shared."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "Set"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "("
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"result"'
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", execResult)"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "            return"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " flyt.DefaultAction, "
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: "nil"
                })]
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "        }),"
                })
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "    )"
                })
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "    "
                })
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "    saveNode "
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: ":="
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " flyt."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "NewNode"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "("
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "        flyt."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "WithPrepFunc"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "("
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "func"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "("
                }), jsx(_components.span, {
                  style: {
                    color: "#E36209",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "ctx"
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: " context"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "Context"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#E36209",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "shared"
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: " *"
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "flyt"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "SharedStore"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ") ("
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "any"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "error"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ") {"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "            result, _ "
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: ":="
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " shared."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "Get"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "("
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"result"'
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ")"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "            return"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " result, "
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: "nil"
                })]
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "        }),"
                })
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "        flyt."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "WithExecFunc"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "("
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "func"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "("
                }), jsx(_components.span, {
                  style: {
                    color: "#E36209",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "ctx"
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: " context"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "Context"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#E36209",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "result"
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: " any"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ") ("
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "any"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "error"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ") {"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "            fmt."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "Printf"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "("
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"Saving result: '
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#F47067"
                  },
                  children: "%d\\n"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"'
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", result)"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "            return"
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: " nil"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ", "
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: "nil"
                })]
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "        }),"
                })
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "    )"
                })
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "    "
                })
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#6A737D",
                    "--shiki-dark": "#768390"
                  },
                  children: "    // Build and run flow"
                })
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "    flow "
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: ":="
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " flyt."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "NewFlow"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "(fetchNode)"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "    flow."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "Connect"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "(fetchNode, flyt.DefaultAction, processNode)"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "    flow."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "Connect"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "(processNode, flyt.DefaultAction, saveNode)"
                })]
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "    "
                })
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "    err "
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: ":="
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " flow."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "Run"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "(context."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "Background"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "(), flyt."
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "NewSharedStore"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "())"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "    if"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " err "
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "!="
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: " nil"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " {"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: "        panic"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "(err)"
                })]
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "    }"
                })
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "}"
                })
              })]
            })
          })
        })
      })]
    }), "\n", jsxs("div", {
      style: {
        marginTop: "3rem"
      },
      children: [jsxs(_components.h2, {
        id: "why-flyt",
        children: ["Why Flyt?", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#why-flyt",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      }), jsx("p", {
        style: {
          fontSize: "1.05rem",
          lineHeight: "1.7",
          color: "var(--vocs-color_text2)",
          marginTop: "1rem"
        },
        children: jsxs(_components.p, {
          children: ["Flyt was inspired by ", jsx("a", {
            href: "https://github.com/The-Pocket/PocketFlow",
            style: {
              color: "var(--vocs-color_link)"
            },
            children: "Pocket Flow"
          }), " and\ndesigned to provide a simple, dependency-free solution for workflow orchestration in Go. Whether you're building\ndata pipelines, API orchestration, or complex business logic, Flyt gives you the tools to create maintainable,\ntestable workflows."]
        })
      }), jsx("h3", {
        style: {
          marginTop: "2rem",
          marginBottom: "1.5rem"
        },
        children: "Perfect for"
      }), jsxs("div", {
        style: {
          display: "grid",
          gridTemplateColumns: "repeat(auto-fit, minmax(280px, 1fr))",
          gap: "1rem"
        },
        children: [jsxs("div", {
          style: {
            padding: "1rem",
            backgroundColor: "var(--vocs-color_backgroundDark)",
            borderRadius: "6px",
            border: "1px solid var(--vocs-color_border)"
          },
          children: [jsx("strong", {
            children: "Data Processing Pipelines"
          }), jsx("p", {
            style: {
              color: "var(--vocs-color_text2)",
              fontSize: "0.9rem",
              marginTop: "0.5rem"
            },
            children: jsx(_components.p, {
              children: "ETL workflows with built-in error handling"
            })
          })]
        }), jsxs("div", {
          style: {
            padding: "1rem",
            backgroundColor: "var(--vocs-color_backgroundDark)",
            borderRadius: "6px",
            border: "1px solid var(--vocs-color_border)"
          },
          children: [jsx("strong", {
            children: "API Orchestration"
          }), jsx("p", {
            style: {
              color: "var(--vocs-color_text2)",
              fontSize: "0.9rem",
              marginTop: "0.5rem"
            },
            children: jsx(_components.p, {
              children: "Chain multiple API calls with automatic retries"
            })
          })]
        }), jsxs("div", {
          style: {
            padding: "1rem",
            backgroundColor: "var(--vocs-color_backgroundDark)",
            borderRadius: "6px",
            border: "1px solid var(--vocs-color_border)"
          },
          children: [jsx("strong", {
            children: "Business Logic Flows"
          }), jsx("p", {
            style: {
              color: "var(--vocs-color_text2)",
              fontSize: "0.9rem",
              marginTop: "0.5rem"
            },
            children: jsx(_components.p, {
              children: "Complex decision trees and state machines"
            })
          })]
        }), jsxs("div", {
          style: {
            padding: "1rem",
            backgroundColor: "var(--vocs-color_backgroundDark)",
            borderRadius: "6px",
            border: "1px solid var(--vocs-color_border)"
          },
          children: [jsx("strong", {
            children: "Batch Operations"
          }), jsx("p", {
            style: {
              color: "var(--vocs-color_text2)",
              fontSize: "0.9rem",
              marginTop: "0.5rem"
            },
            children: jsx(_components.p, {
              children: "Process large datasets concurrently"
            })
          })]
        }), jsxs("div", {
          style: {
            padding: "1rem",
            backgroundColor: "var(--vocs-color_backgroundDark)",
            borderRadius: "6px",
            border: "1px solid var(--vocs-color_border)"
          },
          children: [jsx("strong", {
            children: "AI/LLM Workflows"
          }), jsx("p", {
            style: {
              color: "var(--vocs-color_text2)",
              fontSize: "0.9rem",
              marginTop: "0.5rem"
            },
            children: jsx(_components.p, {
              children: "Build agent systems and conversation flows"
            })
          })]
        })]
      })]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
function _missingMdxReference(id, component) {
  throw new Error("Expected " + (component ? "component" : "object") + " `" + id + "` to be defined: you likely forgot to import, pass, or provide it.");
}
export {
  MDXContent as default,
  frontmatter
};
